core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -303,6 +331,263 @@ public static Properties filterAndRemovePrefix(Map<String, String> properties, S
+   * @return the SQL statement.
+   */
+  public static String listTablesOrViewsSqlStatement(boolean table) {
+    String tableOrViewTableName = table ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String tableOrViewNamespace = table ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    return String.format(LIST_TABLES_OR_VIEWS_SQL, tableOrViewTableName, tableOrViewNamespace);
+  }
+
+  /**
+   * Create SQL statement to rename a table or view.
+   *
+   * @param table true to get the SQL statement for a table, false for a view.
+   * @return the SQL statement.
+   */
+  public static String renameTableOrViewSqlStatement(boolean table) {
+    String tableOrViewTableName = table ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String tableOrViewNamespace = table ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    String tableOrViewName = table ? TABLE_NAME : VIEW_NAME;
+    return String.format(
+        RENAME_TABLE_OR_VIEW_SQL,
+        tableOrViewTableName,
+        tableOrViewNamespace,
+        tableOrViewName,
+        tableOrViewNamespace,
+        tableOrViewName);
+  }
+
+  /**
+   * Create SQL statement to delete a table or view.
+   *
+   * @param table true to get the SQL statement for a table, false for a view.
+   * @return the SQL statement.
+   */
+  public static String dropTableOrViewSqlStatement(boolean table) {
+    String tableOrViewTableName = table ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String tableOrViewNamespace = table ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    String tableOrViewName = table ? TABLE_NAME : VIEW_NAME;
+    return String.format(
+        DROP_TABLE_OR_VIEW_SQL, tableOrViewTableName, tableOrViewNamespace, tableOrViewName);
+  }
+
+  /**
+   * Create SQL statement to create a table or view.
+   *
+   * @param table true to get the SQL statement for a table, false for a view.
+   * @return the SQL statement.
+   */
+  public static String doCommitCreateTableOrViewSqlStatement(boolean table) {
+    String tableOrViewTableName = table ? CATALOG_TABLE_NAME : CATALOG_VIEW_NAME;
+    String tableOrViewNamespace = table ? TABLE_NAMESPACE : VIEW_NAMESPACE;
+    String tableOrViewName = table ? TABLE_NAME : VIEW_NAME;
+    return String.format(
+        DO_COMMIT_CREATE_TABLE_OR_VIEW_SQL,
+        tableOrViewTableName,
+        tableOrViewNamespace,
+        tableOrViewName);
+  }
+
+  /**
+   * Update a table or view.
+   *
+   * @param table true to update a table, false to update a view.
+   * @param connections the {@link JdbcClientPool} to connect to the database.
+   * @param catalogName the catalog name.
+   * @param tableOrViewIdentifier the table or view identifier.
+   * @param newMetadataLocation the new metadata location.
+   * @param oldMetadataLocation the old metadata location.
+   * @return the number of updated record.
+   * @throws SQLException if a SQL error happens on the database.
+   * @throws InterruptedException interrupt the SQL query execution.
+   */
+  public static int updateTableOrView(
```

Comment:
These public interfaces can also be `updateTable` and `updateView` and internally it can call a private method `updateContent` with the boolean flag.

Reply:
similarly for below methods.